packages/wallet/src/redux/protocols/advance-channel/reducer.ts

```diff
@@ -149,22 +147,60 @@ function initializeWithNewChannel(
   }
 }
 
-const channelUnknownReducer: ProtocolReducer<states.AdvanceChannelState> = (
-  protocolState: states.ChannelUnknown,
-  sharedData,
-  action: CommitmentsReceived,
-) => {
-  const { ourIndex, privateKey, commitmentType } = protocolState;
-  const channelId = getChannelId(action.signedCommitments[0].commitment);
+function initializeWithExistingChannel(
+  processId,
+  sharedData: Storage,
+  initializeChannelArgs: OngoingChannelArgs,
+) {
+  const { channelId, ourIndex } = initializeChannelArgs;
+  const channel = getChannel(sharedData.channelStore, channelId);
+  if (!channel) {
+    throw new Error('Channel not stored');
```

Comment:
Should we standardize this error for the scenario of channel expected but missing?

Reply:
This check isn't necessary, now that `getChannel` always returns a `ChannelState`.